### PR TITLE
Ejercicio 2

### PR DESCRIPTION
**Pull request by Joelit0**
Exercise 2 is similar to Exercise 1 but is now able to receive a directory path. Now the user can put a directory path that contains subfolders or a simple file.

### CODE REVIEW

File: Ejercicio5b.rb [hunk 1]
```diff
@@ -2,155 +2,143 @@
...
+  file_or_route= gets.chomp.capitalize
+  puts "========================================================================================"
+  
+  if file_or_route == "Archivo"
+    puts "---Ingrese el nombre del archivo que quiere subir como un Gist de GitHub--- " 
     puts "========================================================================================"
-    puts "---Ingrese si quiere subir un archivo o un ruta---"
-    puts "========================================================================================"
-    file_or_route= gets.chomp.capitalize
+    filename= gets.chomp
```
Comment (pazcuturi) [inline Ejercicio5b.rb:20]: add missing space please

File: Ejercicio5b.rb [hunk 2]
```diff
@@ -2,155 +2,143 @@
...
-                break if try_again == "No"
-            end
-        else
-            puts "Su archivo no existe"
-            puts "========================================================================================"
-        end
-
-    elsif file_or_route == "Ruta"
-        puts "---Ingrese la ruta del directorio que desea subir como un gist de Github---"
+      if response_status== "201"
```
Comment (pazcuturi) [inline Ejercicio5b.rb]: add missing space please

File: Ejercicio5b.rb [hunk 3]
```diff
@@ -2,155 +2,143 @@
...
-            rescue SocketError => se
-                puts "Ha ocurrido un error de conexión. Quiere intentar nuevamente?"
-                try_again = gets.chomp.capitalize
-                break if try_again == "No"
-            end
+      only_files = Dir.glob("#{route}"+'/**/*').reject do |path|
+        File.directory?(path)
+      end
+      
+      @bar ={ :description => description,:public => state,:files =>{} }
```
Comment (pazcuturi) [inline Ejercicio5b.rb:86]: add missing space please

File: Ejercicio5b.rb [hunk 4]
```diff
@@ -2,155 +2,143 @@
...
+      req_options = { use_ssl: uri.scheme == "https" }
+      
+      begin
+        response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
+          http.request(request)
+      end
+      
+      json = response.body
+      parsed = JSON.parse(json)
+      response_status = "#{response.code}"
```
Comment (pazcuturi) [inline Ejercicio5b.rb]: Why interpolate this when you can use an integer instead?

File: Gist_Request.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    @state = state
+    @content = content
+    @description = description
+  end
+
+  def post()
+    uri = URI.parse(GITHUB_URL)
+    request = Net::HTTP::Post.new(uri)
+    request.basic_auth(ENV['YOUR_USERNAME'], ENV['YOUR_TOKEN'])
+    request.body = JSON.dump({ "description" => @description,"public" => @state,"files" => { @filename => { "content" => @content }}})
```
Comment (pazcuturi) [inline Gist_Request.rb]: add spaces after comma please

File: Gist_Request.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    req_options = { use_ssl: uri.scheme == "https" }
+    
+    begin
+      response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
+        http.request(request)
+    end
+    
+    json = response.body
+    parsed = JSON.parse(json)
+    @response_status = response.code
```
Comment (pazcuturi) [inline Gist_Request.rb]: Same here, why use string interpolation when you can use the integer instead?

File: Main.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
+require_relative 'Gist_Attributes'
+require_relative 'Gist_Request'
+
+loop do
+  puts "Ingrese el nombre de su archivo"
+  filename = gets.chomp
+  puts "Ingrese una descripción"
+  description = gets.chomp
+  puts "Quieres que sea público?"
+  state= gets.chomp.capitalize
```
Comment (pazcuturi) [inline Main.rb]: add missing space please

File: Ejercicio5b.rb [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      puts "========================================================================================"
+      state = gets.chomp.capitalize
+
+      if state == "Si" 
+        state = true;
+      elsif state == "No"
+        state = false;
+      else
+        puts "Responde Si o No..."
+      end
```
Comment (jrgonh) [inline Ejercicio5b.rb]: state = state != "No"

File: Ejercicio5b.rb [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    puts "---Ingrese el nombre del archivo que quiere subir como un Gist de GitHub--- " 
+    puts "========================================================================================"
+    filename= gets.chomp  
+    puts "========================================================================================"
+
+    if File.exist?(filename)   
+      puts "---Añade una descripción---"
+      puts "========================================================================================"
+      description = gets.chomp
+      puts"---Quieres que tu repositorio sea publico?---"
```
Comment (jrgonh) [inline Ejercicio5b.rb]: ---Quieres que tu repositorio sea público? Si/No, por defecto el repositorio será público---

File: Ejercicio5b.rb [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+      begin
+        response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
+          http.request(request)
+      end
+          
+      json = response.body
+      parsed = JSON.parse(json)
+      response_status = response.code
+
+      if response_status== "201"
```
Comment (jrgonh) [inline Ejercicio5b.rb]: HTTP_STATUS_CREATED = "201"

File: Ejercicio5b.rb [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+      state = gets.chomp.capitalize
+      puts "========================================================================================"
+
+      if state == "Si" 
+        state = true;
+      elsif state == "No"
+        state = false;
+      else
+        puts "Responde si o no..."
+      end
```
Comment (jrgonh) [inline Ejercicio5b.rb]: same as above

File: Ejercicio5b.rb [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+        File.directory?(path)
+      end
+      
+      @bar ={ :description => description,:public => state,:files =>{} }
+      
+      only_files.each do |f|
+        @name = Pathname.new(f).basename 
+        puts @name
+
+        open(f, "r:UTF-8") do |file| # Leer el contenido del archivo
```
Comment (jrgonh) [inline Ejercicio5b.rb]: comments in english

File: Ejercicio5b.rb [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+      begin
+        response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
+          http.request(request)
+      end
+      
+      json = response.body
+      parsed = JSON.parse(json)
+      response_status = response.code
+  
+      if response_status == "201"
```
Comment (jrgonh) [inline Ejercicio5b.rb]: use constants

File: Ejercicio5b.rb [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+      rescue SocketError => se
+        puts "Ha ocurrido un error de conexión. Quiere intentar nuevamente?"
+        try_again = gets.chomp.capitalize
+        break if try_again == "No"
+      end
+    else
+      puts "---Su ruta no existe---"
+    end
+  end
+end
```
Comment (jrgonh) [inline Ejercicio5b.rb]: add newline

**Review by jrgonh — CHANGES_REQUESTED** [review]

**Review by jrgonh — APPROVED** [review]

**Pull request merged**